packages/repack/src/webpack/utils/getResolveOptions.ts

```diff
@@ -17,35 +35,83 @@
+ *         enablePackageExports: false,
+ *         preferNativePlatform: true
+ *       }),
  *     },
  *   };
  * };
  * ```
  */
-export function getResolveOptions(platform: string) {
+
+export function getResolveOptions(platform: string, options?: ResolveOptions) {
+  const preferNativePlatform = options?.preferNativePlatform ?? true;
+  const enablePackageExports = options?.enablePackageExports ?? false;
+
+  let extensions = ['.ts', '.js', '.tsx', '.jsx', '.json'];
+
+  const platformExtensions = [
+    `.${platform}.ts`,
+    `.${platform}.js`,
+    `.${platform}.tsx`,
+    `.${platform}.jsx`,
+  ];
+
+  const nativeExtensions = [
+    '.native.ts',
+    '.native.js',
+    '.native.tsx',
+    '.native.jsx',
+  ];
+
+  let conditionNames: string[];
+  let exportsFields: string[];
+
+  if (enablePackageExports) {
+    /**
+     * Match what React Native uses in @react-native/metro-config.
+     * Order of conditionNames doesn't matter.
+     * Order inside of target package.json's `exports` field matters.
+     */
+    conditionNames = ['require', 'import', 'react-native'];
+    exportsFields = ['exports'];
+  } else {
+    conditionNames = [];
+    exportsFields = [];
+    extensions = [
+      platformExtensions,
+      preferNativePlatform ? nativeExtensions : [],
+      extensions,
+    ].flat();
+  }
+
+  /**
+   * Match what React Native uses in @react-native/metro-config.
+   * First entry takes precedence.
+   *
+   * Reference: https://github.com/facebook/react-native/blob/main/packages/metro-config/src/index.flow.js
+   */
   return {
     /**
-     * Match what React Native packager supports.
-     * First entry takes precedence.
+     * Reference: Webpack's [configuration.resolve.mainFields](https://webpack.js.org/configuration/resolve/#resolvemainfields)
      */
     mainFields: ['react-native', 'browser', 'main'],
+    /**
+     * Reference: Webpack's [configuration.resolve.aliasFields](https://webpack.js.org/configuration/resolve/#resolvealiasfields)
+     */
     aliasFields: ['react-native', 'browser', 'main'],
-    conditionNames: ['default', 'require'],
-    extensions: [
-      `.${platform}.ts`,
-      `.${platform}.js`,
-      `.${platform}.tsx`,
-      `.${platform}.jsx`,
-      '.native.ts',
-      '.native.js',
-      '.native.tsx',
-      '.native.jsx',
-      '.ts',
-      '.js',
-      '.tsx',
-      '.jsx',
-      '.json',
-    ],
+    /**
+     * Reference: Webpack's [configuration.resolve.conditionNames](https://webpack.js.org/configuration/resolve/#resolveconditionnames)
+     */
+    conditionNames: conditionNames,
+    /**
+     * Reference: Webpack's [configuration.resolve.exportsFields](https://webpack.js.org/configuration/resolve/#resolveexportsfields)
+     */
+    exportsFields: exportsFields,
+    /**
+     * Reference: Webpack's [configuration.resolve.extensions](https://webpack.js.org/configuration/resolve/#resolveextensions)
+     */
+    extensions: extensions,
```

Comment:
nit:
```suggestion
    conditionNames,
    /**
     * Reference: Webpack's [configuration.resolve.exportsFields](https://webpack.js.org/configuration/resolve/#resolveexportsfields)
     */
    exportsFields,
    /**
     * Reference: Webpack's [configuration.resolve.extensions](https://webpack.js.org/configuration/resolve/#resolveextensions)
     */
    extensions,
```

Reply:
This is there only for proper displaying of the typedoc/rendering API doc page - without this it renders incorrectly afaik.